sql/mozfun/norm/product_info/metadata.yaml

```diff
@@ -0,0 +1,54 @@
+friendly_name: Normalize Product Info
+description: |
+  Returns a normalized `product` name and a `canonical_name` for a product
```

Comment:
@gkabbz Please take a look at this documentation for `mozfun.norm.product_info`. My intention is that this becomes the canonical reference for naming of mobile applications in telemetry. The function logic is generated from the markdown table.